lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -358,23 +363,35 @@ def _on_update_status(self, event: UpdateStatusEvent):
             self.unit.status = BlockedStatus(" - ".join(missing_sys_reqs))
             return
 
+        try:
+            # Retrieve the nodes of the cluster, needed to configure this node
+            nodes = self._get_nodes(False)
+            # Set the configuration of the node
+            self._set_node_conf(nodes)
+        except OpenSearchHttpError:
+            pass
+
+        self._reconfigure_and_restart_unit_if_needed()
```

Comment:
This will unfortunately not have any effect on the non leader units, as: 
- `nodes_config` is only set from the leader
- the current logic is roles bound

Please extend the `_reconfigure_and_restart_unit_if_needed` method with a new argument: `conf_from_leader: bool = True`

i.e:
```
def _reconfigure_and_restart_unit_if_needed(self, conf_from_leader: bool = True):
    """Reconfigure the current unit if a new config was computed for it, then restart."""
    if conf_from_leader:
        nodes_config = self.peers_data.get_object(Scope.APP, "nodes_config")
        if not nodes_config:
            return

         new_node_conf = nodes_config.get(self.unit_name)
    else:
         new_node_conf = # recompute the conf

    .....
```

Reply:
You can restrain this new logic to only checking `network_host`

Reply:
Actually much easier for now, since we'll implement a config reconciliation later: create a new method in `opensearch_config.py`: `update_host_if_needed(host: str)` that loads the config, checks if passed host is the list of the stored `network.host`, if yes return False otherwise replaces the previous one with the passed one and return True.
If return True, emit a restart unit
(don't forget to add back `["_site_"]` to the liust of hosts)